xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-api/src/main/java/org/xwiki/search/solr/SolrEntityMetadataExtractor.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.search.solr;
+
+import org.apache.solr.common.SolrInputDocument;
+import org.xwiki.component.annotation.Role;
+import org.xwiki.stability.Unstable;
+
+/**
+ * An extension point to be called when an entity is indexed for the Solr search core.
+ * 
+ * @param <E> the type of the parsed entity
+ * @version $Id$
+ * @since 14.8RC1
+ */
+@Role
+@Unstable
+public interface SolrEntityMetadataExtractor<E>
```

Comment:
That's a new role you provide, but there's no component implementing it yet? Or I missed it?

Reply:
You are not missing something. I used to use this new system to implement link indexing from the refactoring module, but link indexing is not part of the core metadata extractor. So it's a new system for extensions which is not use by XWiki Standard, I did not have the heart to remove it (it would be very nice to remove a few hacks in https://extensions.xwiki.org/xwiki/bin/view/Extension/Solr%20Security/).